pkg/secrets/strategy.go

```diff
@@ -130,18 +118,128 @@ func (s Source) MarshalYAML() (interface{}, error) {
 	return s.MarshalRaw(), nil
 }
 
-type StrategyList []SourceMap
+// SourceMapList is a list of mappings that can be access via index or the item name.
+type SourceMapList []SourceMap
 
-func (l StrategyList) Less(i, j int) bool {
+func (l SourceMapList) Less(i, j int) bool {
 	return l[i].Name < l[j].Name
 }
 
-func (l StrategyList) Swap(i, j int) {
+func (l SourceMapList) Swap(i, j int) {
 	tmp := l[i]
 	l[i] = l[j]
 	l[j] = tmp
 }
 
-func (l StrategyList) Len() int {
+func (l SourceMapList) Len() int {
 	return len(l)
 }
+
+// HasName determines if the specified name is defined in the set.
+func (l SourceMapList) HasName(name string) bool {
+	_, ok := l.GetByName(name)
+	return ok
+}
+
+// GetByName returns the resolution strategy for the specified name and a bool indicating if it was found.
+func (l SourceMapList) GetByName(name string) (SourceMap, bool) {
+	for _, item := range l {
+		if item.Name == name {
+			return item, true
+		}
+	}
+
+	return SourceMap{}, false
+}
+
+// GetResolvedValue returns the resolved value of the specified name and a bool indicating if it was found.
+// You must resolve the value before calling, it does to do resolution for you.
+func (l SourceMapList) GetResolvedValue(name string) (interface{}, bool) {
+	item, ok := l.GetByName(name)
+	if ok {
+		return item.ResolvedValue, true
+	}
+
+	return nil, false
+}
+
+// GetResolvedValues converts the items to a map of key/value pairs, with the values represented as CNAB-compatible strings
+func (l SourceMapList) GetResolvedValues() map[string]string {
+	values := make(map[string]string, len(l))
+	for _, item := range l {
+		values[item.Name] = item.ResolvedValue
+	}
+	return values
+}
+
+// GetTypedResolvedValues converts the items to a map of key/value pairs, where the value is the type defined by the bundle.
+// Validation errors are accumulated, and parameters are always passed back along with a validation error.
+// If you care about valid parameters, check the error, but otherwise you can use this for display purposes and not have it blow up with no results.
+func (l SourceMapList) GetTypedResolvedValues(bun cnab.ExtendedBundle) (map[string]interface{}, error) {
```

Comment:
I copied this from elsewhere in porter while refactoring and just realized that the functionality was only tested higher up in the code, but never as a unit test for the function itself. Adding one now...

Reply:
Oops, never mind this needs to be defined directly on ParameterSet, and not on the generic SourceMapList. I'll remove it from this PR.